sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentPatchResource.java

```diff
@@ -0,0 +1,223 @@
+  @Parameters(
+      value = {
+        @Parameter(
+            name = "namespace",
+            ref = OpenApiConstants.Parameters.NAMESPACE,
+            description = "The namespace of the document."),
+        @Parameter(
+            name = "collection",
+            ref = OpenApiConstants.Parameters.COLLECTION,
+            description = "The collection of the document (Will be created if it does not exist)."),
+        @Parameter(
+            name = "document-id",
+            ref = OpenApiConstants.Parameters.DOCUMENT_ID,
+            description = "The ID of the document that you'd like to patch"),
+        @Parameter(
+            name = "ttl-auto",
+            ref = OpenApiConstants.Parameters.TTL_AUTO,
+            description =
+                "Include this to make the TTL match that of the parent document. Requires read-before-write if set to true"),
+        @Parameter(
+            name = "profile",
+            ref = OpenApiConstants.Parameters.PROFILE,
+            description = "Include profiling information from execution."),
+      })
+  @APIResponses(
+      value = {
+        @APIResponse(
+            responseCode = "200",
+            description =
+                "Document updated or created with the provided document-id. The document-id will be returned.",
+            content = {
+              @Content(
+                  schema =
+                      @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                          implementation = SimpleResponseWrapper.class,
+                          properties =
+                              @SchemaProperty(name = "documentId", type = SchemaType.STRING)))
+            }),
+        @APIResponse(
+            responseCode = "404",
+            description = "Not found.",
+            content =
+                @Content(
+                    examples = {
+                      @ExampleObject(ref = OpenApiConstants.Examples.NAMESPACE_DOES_NOT_EXIST)
+                    },
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @PATCH
+  public Uni<RestResponse<Object>> patchDocument(
+      @PathParam("namespace") String namespace,
+      @PathParam("collection") String collection,
+      @PathParam("document-id") String documentId,
+      @QueryParam("ttl-auto") boolean ttlAuto,
+      @QueryParam("profile") boolean profile,
+      @NotNull JsonNode body) {
+    ExecutionContext context = ExecutionContext.create(profile);
+    Uni<Schema.CqlTable> table = tableManager.ensureValidDocumentTable(namespace, collection);
```

Comment:
So can patch be done without an extsing document? If not, then we should have `getValidDocumentTable` here.. Not sure what's the logic we support atm..

Reply:
Yeah - in v1 a patch can be done without the document/collection existing, and in this case it behaves trivially just like a PUT